cluster/method/kmeans.py

```diff
@@ -166,3 +169,103 @@ def initialise_clusters(self, input_, clustercount):
         for item in input_:
             self.__clusters[count % clustercount].append(item)
             count += 1
+
+
+    def HDgetclusters(self, count,  max_iterations):
+        """
+        Generates *count* clusters.
+
+        :param count: The amount of clusters that should be generated.  count
+            must be greater than ``1``.
+        :raises ClusteringError: if *count* is out of bounds.
+        """
+
+        # only proceed if we got sensible input
+        if count <= 1:
+            raise ClusteringError("When clustering, you need to ask for at "
+                                  "least two clusters! "
+                                  "You asked for %d" % count)
+
+        # return the data straight away if there is nothing to cluster
+        if (self.__data == [] or len(self.__data) == 1 or
+                count == self.__initial_length):
+            return self.__data
+
+        # It makes no sense to ask for more clusters than data-items available
+        if count > self.__initial_length:
+            raise ClusteringError(
+                "Unable to generate more clusters than "
+                "items available. You supplied %d items, and asked for "
+                "%d clusters." % (self.__initial_length, count))
+
+        self.initialise_clusters(self.__data, count)
+
+        items_moved = True  # tells us if any item moved between the clusters,
+                            # as we initialised the clusters, we assume that
+                            # is the case
+
+        iteration=0
+        #asi no, no obligar a hacer iteraciones, lo hago segun dice el algoritmo
+        #pero si llego a iteraciones paro, si termino antes de llegar, mejor
+        while items_moved is True:
+            items_moved = False
+            print "iterating",iteration
```

Comment:
This should be replaced with a logging statement. Otherwise it will cause unwanted "noise" on `stdout` for everyone using this method which is even impossible to silence without modifying the code.

Reply:
ok, you are right